pkg/kubectl/polymorphichelpers/attachablepodforobject.go

```diff
@@ -21,23 +21,16 @@ import (
 	"sort"
 	"time"
 
-	"k8s.io/api/core/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/cli-runtime/pkg/genericclioptions"
 	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
-	api "k8s.io/kubernetes/pkg/apis/core"
-	apiv1 "k8s.io/kubernetes/pkg/apis/core/v1"
 	"k8s.io/kubernetes/pkg/controller"
 )
 
 // attachablePodForObject returns the pod to which to attach given an object.
 func attachablePodForObject(restClientGetter genericclioptions.RESTClientGetter, object runtime.Object, timeout time.Duration) (*corev1.Pod, error) {
 	switch t := object.(type) {
-	case *api.Pod:
```

Comment:
How do we know this is never happening?

Reply:
This function is only called in three places:

1) attach.go: called here with an object created from the builder with external scheme. Therefore, the object must be an external version.
2) portforward.go: called here with an object created from the builder with external scheme. Therefore, the object must be an external version.
3) run.go: again, the object used as a parameter to this function can not be internal. There is a specific function called to ensure this object is external version.

Reply:
And we also know they can be of only one version?